### PR TITLE
#939 Skeleton for Viber radar

### PR DESCRIPTION
**Pull request by carlosmiranda**
#939: Created basic framework for Viber radar. Added puzzles for further implementation.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1001](https://github.com/zerocracy/farm/issues/1001) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1001](https://github.com/zerocracy/farm/issues/1001) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1290741272); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

**Comment by carlosmiranda** [comment]
@g4s8 ping

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: src/main/java/com/zerocracy/radars/viber/Reaction.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.io.IOException;
+
+/**
+ * Viber message reaction.
+ *
+ * @author Carlos Miranda (miranda.cma@gmail.com)
+ * @version $Id$
+ * @since 0.22
+ */
+public interface Reaction {
```
Comment (g4s8) [inline src/main/java/com/zerocracy/radars/viber/Reaction.java]: @carlosmiranda maybe this interface can be package-protected, not public?

File: src/main/java/com/zerocracy/radars/viber/TkViber.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ *  events, we should route them to ReProfile. For the rest we should at
+ *  least be able to return a 200 OK response to Viber. When fully implemented
+ *  this should be added to Main.java so that it can be called. See
+ *  https://developers.viber.com/docs/api/rest-bot-api/
+ */
+public final class TkViber implements Take {
+
+    @Override
+    public Response act(final Request request) throws IOException {
+        throw new UnsupportedOperationException("Not yet implemented.");
```
Comment (g4s8) [inline src/main/java/com/zerocracy/radars/viber/TkViber.java]: @carlosmiranda I think it's better to return `501 NOT_IMPLEMENTED` status code for not implemented http endpoints.

File: src/main/java/com/zerocracy/radars/viber/VbBot.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+package com.zerocracy.radars.viber;
+
+/**
+ * Viber bot.
+ *
+ * @author Carlos Miranda (miranda.cma@gmail.com)
+ * @version $Id$
+ * @since 0.22
+ * @todo #939:30min Implement Viber bot. We should be able to send messages via
+ * the Viber REST API. There are other operations but I think that is the only
```
Comment (g4s8) [inline src/main/java/com/zerocracy/radars/viber/VbBot.java]: @carlosmiranda incorrect indentation here (should be 2 spaces), pdd won't parse this

**Comment by carlosmiranda** [comment]
@g4s8 fixed, please check

**Comment by carlosmiranda** [comment]
@ping

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by g4s8** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @g4s8[/z](https://www.0crat.com/u/g4s8), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1001](https://github.com/zerocracy/farm/issues/1001) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@g4s8 According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

One out of three found issues is cosmetic. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by g4s8** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@g4s8 Thanks!

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**